core/primitives/src/shard_layout.rs

```diff
@@ -695,6 +695,12 @@ impl ShardLayout {
         self.shard_ids().map(|shard_id| ShardUId::from_shard_id_and_layout(shard_id, self))
     }
 
+    pub fn shard_infos(&self) -> impl Iterator<Item = ShardInfo> + '_ {
```

Comment:
nit: I was curious about this `'_`, and actually `shard_ids()` doesn't seem to need it